packages/smart/contracts/turbo/AMMFactory.sol

```diff
@@ -45,33 +56,45 @@ contract AMMFactory is BNum {
         uint256 price
     );
 
-    constructor(BFactory _bFactory, uint256 _fee) {
+    constructor(
+        BFactory _bFactory,
+        MasterChef _masterChef,
+        uint256 _fee
+    ) {
         bFactory = _bFactory;
+        masterChef = _masterChef;
         fee = _fee;
     }
 
+    function approveMarketFactory(address _marketFactory) external onlyOwner returns (bool) {
+        approvedMarketFactories[_marketFactory] = true;
+        return true;
```

Comment:
Why return a bool? This is a common pattern but as far as I can tell it's only useful when you have to adhere to some interface designed before people realized that everyone just uses reverts instead of returning a status code.

Reply:
Good call.